src/OpenDiffix.Core/Anonymizer.fs

```diff
@@ -356,16 +358,21 @@ let sum (anonParams: AnonymizationParams) (anonContext: AnonymizationContext) (p
     let anonymizedSumOnNonEmpty =
       Array.choose id
       >> function
-        | [||] -> 0.0
-        | nonEmpty -> anonymizedSum nonEmpty |> fst
+        | [||] -> (0.0, 0.0)
+        | nonEmpty -> anonymizedSum nonEmpty
 
     // Using `anonymizedSum` separately for positive and negative, we ensure that we pick the appropriate
     // amount of flattening and noise for each leg, and only later combine the results.
-    let positive = anonymizedSumOnNonEmpty byAidPositive
-    let negative = anonymizedSumOnNonEmpty byAidNegative
+    let (positive, positiveNoiseSD) = anonymizedSumOnNonEmpty byAidPositive
+    let (negative, negativeNoiseSD) = anonymizedSumOnNonEmpty byAidNegative
+    let noiseSD = Math.Sqrt(positiveNoiseSD ** 2.0 + negativeNoiseSD ** 2.0)
```

Comment:
this is a formula for _independent_ variables, and currently my best shot, but I need to give it some more thought.

`positive` and `negative` aren't independent (the noisy `AnonymizedSums`, that is), so I guess their noise isn't either.

Still TBD, I'll make a self-note to kick off a discussion, unless someone has a quick argument I'm missing here.

Reply:
This makes sense, but shouldn't it also contain a check for the `NotEnoughAIDVs` case?

Reply:
I now saw the previous check that requires both (positive and negative) partial results to be valid. This is not correct: it is enough to have one valid partial result in order to return a final valid result.

Reply:
it does, it's several lines above. If neither positive nor negative batch have enough AIDVs for a particular AID instance, it will return `AnonymizedResult.NotEnoughAIDVs`. In `Aggregator.fs` this will get translated to `Null`.

At this line here, SDs of both batches are the maximal ones where there were enough AIDVs.

Reply:
> I now saw the previous check that requires both (positive and negative) partial results to be valid. This is not correct: it is enough to have one valid partial result in order to return a final valid result.

No, it fails if both positive and negative are invalid. If there are no negative values whatsoever (typical `age` column), it will succeed, if all AID instances have enough AIDVs. Also the `negative` will be 0 and its noise will be 0.0.